### DISCUSSION TITLE
`reproject()` and `reproject_match()` with extra dimensions?

### DISCUSSION THREAD

**Discussion by jamaa** [corteva/rioxarray#649] [Q&A] [answered]
I have data containing extra dimensions besides x, y and time. In one case, the extra dimension is "number", denoting the ensemble member number of a forecast. I would like to reproject this data using `reproject()` and `reproject_match()`. However, I get this error message:

> TooManyDimensions: Only 2D and 3D data arrays supported.

The docs state that 

> Only 2D/3D arrays with dimensions ‘x’/’y’ are currently supported. Others are appended as is.

I wonder why this restriction is in place? After all, the reprojection should only affect the x and y dimensions, all other dimensions should be unaffected and can be kept as-is?

I am thinking of using the following workaround:
* select a single ensemble member and drop the dimension "number", giving me a DataArray with only x, y and time
* perform the reprojection
* re-add the "number" dimension and fill it with the original value
* repeat for all ensemble members and then merge them back to one Dataset

Is there a better way to do this?

**Comment by snowman2** [comment, marked as answer]
> I wonder why this restriction is in place?

That is what has been implemented in the code. It isn't because higher dimensionality couldn't be supported, just that it isn't supported with the current implementation.

> Is there a better way to do this?

What you propose sounds like it should work.

**Reply by jamaa** [reply]
thanks for the reply, my workaround did indeed work
